partiql-parser/src/test/kotlin/org/partiql/parser/internal/PartiQLParserDDLTests.kt

```diff
@@ -521,10 +596,284 @@ class PartiQLParserDDLTests {
+                        listOf(
+                            tableDefinitionAttribute(
+                                identifierSymbol("a", Identifier.CaseSensitivity.INSENSITIVE),
+                                Type.Struct(
+                                    listOf(
+                                        Type.Struct.Field(
+                                            identifierSymbol("b", Identifier.CaseSensitivity.INSENSITIVE),
+                                            Type.Int2(),
+                                            emptyList(),
+                                            false,
+                                            "comment on struct field"
+                                        )
+                                    )
+                                ),
+                                emptyList(),
+                                false,
+                                null
+                            )
+                        ),
+                        emptyList()
+                    ),
+                    null,
+                    emptyList()
+                )
+            ),
+
+            SuccessTestCase(
+                "CREATE TABLE with comment on struct which has comment on field",
+                """
+                    CREATE TABLE tbl (
+                        a STRUCT<b : INT2 COMMENT 'comment on inner level'> COMMENT 'comment on top level'
+                    )
+                """.trimIndent(),
+                ddlOpCreateTable(
+                    identifierSymbol("tbl", Identifier.CaseSensitivity.INSENSITIVE),
+                    tableDefinition(
+                        listOf(
+                            tableDefinitionAttribute(
+                                identifierSymbol("a", Identifier.CaseSensitivity.INSENSITIVE),
+                                Type.Struct(
+                                    listOf(
+                                        Type.Struct.Field(
+                                            identifierSymbol("b", Identifier.CaseSensitivity.INSENSITIVE),
+                                            Type.Int2(),
+                                            emptyList(),
+                                            false,
+                                            "comment on inner level"
+                                        )
+                                    )
+                                ),
+                                emptyList(),
+                                false,
+                                "comment on top level"
+                            )
+                        ),
+                        emptyList()
+                    ),
+                    null,
+                    emptyList()
+                )
+            ),
+
+            // Partition BY
+            SuccessTestCase(
+                "CREATE TABLE with Partition by single attribute",
+                """
+                    CREATE TABLE tbl
```

Comment:
Shouldn't this be syntax error?

Reply:
I think this should be semantic error, i.e., referring a non-existing attribute in Partition By expression.

Reply:
But is the following in a valid syntax considering it lacks the parenthesis and column definitions?

```sql
CREATE TABLE tbl PARTITION BY (a)
```
See:
https://www.db-fiddle.com/f/6PztiiVM6uEzpvdWnMyKX7/0

Reply:
I think so because 
```
CREATE TABLE tbl
```
has been a valid syntax for us. 

Based on this I am treating the grammar as three parts: 

```
CREATE TABLE tbl  -- header
(a INT2) -- data type definition
PARTITION BY (a) -- table extension
```